src/python/pants/python/python_setup.py

```diff
@@ -255,6 +391,16 @@ def get_pyenv_paths(env: Environment, *, pyenv_local: bool = False) -> List[str]
         return paths
 
 
+def get_asdf_dir(env: Environment) -> PurePath | None:
+    """See https://asdf-vm.com/#/core-configuration?id=environment-variables."""
+    asdf_dir = env.get("ASDF_DIR", env.get("ASDF_DATA_DIR"))
```

Comment:
These environment variables have not been include-listed into the `Environment` that will be exposed to the `@rules` which call `PythonSetup.interpreter_search_paths`: the primary caller we care about is `find_pex_pythons`: https://github.com/pantsbuild/pants/blob/b0b7b0b914fd9df86be64e0948a87a01ccbce528/src/python/pants/backend/python/util_rules/pex_environment.py#L134-L136.

Without doing so, this code will never observe those variables, because they will be filtered out.

Reply:
Unless I've misunderstood, I've added the requested variables to the whitelist so that it will function as expected. Is there more that needs to be done?

Reply:
Not that I know of off the top of my head, but: actually trying it on your machine and confirming that you're seeing it working is your best bet. I don't have ASDF installed (yet).